nodejs/lancedb/arrow.ts

```diff
@@ -591,10 +617,19 @@ function inferType(
       return undefined;
     }
     // Try to automatically detect embedding columns.
-    if (valueType instanceof Float && path[path.length - 1] === "vector") {
-      // We default to Float32 for vectors.
-      const child = new Field("item", new Float32(), true);
-      return new FixedSizeList(value.length, child);
+    if (nameSuggestsVectorColumn(path[path.length - 1])) {
+      // Check if values are integers for type determination
+      const isIntegerVector = isAllIntegers(value);
```

Comment:
Because Javascript lacks real integers, I think this could get false positives. Especially if the user provides all zeros for one of the sample values. Instead, let's only assume an integer vector if:

1. `nameSuggestsVectorColumn()`
2. `value` is a `Uint8Array`